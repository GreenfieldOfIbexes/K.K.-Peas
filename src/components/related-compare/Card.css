.deck .product-card {
	width: 270px;
	min-width: 270px;
	height: 420px;
	overflow: hidden;
	display: flex;
	flex-direction: column;
	position: relative;
	justify-content: unset;
	align-items: unset;
	box-shadow: 0 0 5px #0000004f;
	margin: 5px;
	border-radius: 5px;
}
.deck p {
	margin-bottom: 0;
}
.deck h2 {
	font-size: 25px;
}
.deck .product-card .product-image {
	min-height: 60%;
	height: 60%;
	width: 100%;
	background-size: cover;
	background-position: center;
	background-repeat: no-repeat;
}
.product-card .bottom-content {
	padding: 10px;
	display: flex;
	flex-direction: column;
	align-items: flex-start;
}
.deck .material-icons {
	position: absolute;
	top: 10px;
	right: 10px;
	cursor: pointer;
}

.pricing {
	float: left;
	display: flex;
}
.pricing > * {
	margin-right: 20px;
}

.crossout-price {
	color: red;
	text-decoration: line-through;
}

.related-product-name {
	cursor: pointer;
}
