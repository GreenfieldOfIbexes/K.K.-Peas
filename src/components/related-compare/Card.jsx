import React from "react";
import Stars from "../R_R/Stars.jsx";
import newMainProduct from "../../actions/newMainProduct.js";
import store from "../../store.js";

//TO-DO: implement sale price formatting
const Card = (props) => {
	return (
		<>
			<img src={props.product.styles.results[0].photos[0].thumbnail_url} />
			<p>{props.product.category}</p>
			<h2>{props.product.name}</h2>
			<p>{"$" + props.product.styles.results[0].original_price} </p>
			<button
				onClick={() => {
					store.dispatch(newMainProduct(props.product.id));
				}}>
				Check it out!
			</button>
			<Stars />
		</>
	);
};

export default Card;
