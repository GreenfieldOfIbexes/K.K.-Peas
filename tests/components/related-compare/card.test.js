import React from "react";
import Card from "../../../src/components/related-compare/Card.jsx";
import renderer from "react-test-renderer";
const { testProduct } = require("../../testProducts.js");

test("Activity card renders correctly", () => {
	// const card = renderer.create(<Card product={testProduct} />).toJSON();
	// expect(card).toMatchSnapshot();
});
