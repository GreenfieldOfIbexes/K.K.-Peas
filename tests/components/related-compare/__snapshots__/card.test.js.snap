// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity card renders correctly 1`] = `
Array [
  <img
    src="https://images.unsplash.com/photo-1501088430049-71c79fa3283e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=300&q=80"
  />,
  <p>
    Jackets
  </p>,
  <h2>
    Camo Onesie
  </h2>,
  <p>
    $140
     
  </p>,
  <div
    className="star-ratings"
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "position": "relative",
      }
    }
    title="3.65 Stars"
  >
    <svg
      className="star-grad"
      style={
        Object {
          "height": "0",
          "position": "absolute",
          "visibility": "hidden",
          "width": "0",
          "zIndex": "0",
        }
      }
    >
      <defs>
        <linearGradient
          id="starGrad691250465249835"
          x1="0%"
          x2="100%"
          y1="0%"
          y2="0%"
        >
          <stop
            className="stop-color-first"
            offset="0%"
            style={
              Object {
                "stopColor": "gold",
                "stopOpacity": "1",
              }
            }
          />
          <stop
            className="stop-color-first"
            offset="65%"
            style={
              Object {
                "stopColor": "gold",
                "stopOpacity": "1",
              }
            }
          />
          <stop
            className="stop-color-final"
            offset="65%"
            style={
              Object {
                "stopColor": "rgb(203, 211, 227)",
                "stopOpacity": "1",
              }
            }
          />
          <stop
            className="stop-color-final"
            offset="100%"
            style={
              Object {
                "stopColor": "rgb(203, 211, 227)",
                "stopOpacity": "1",
              }
            }
          />
        </linearGradient>
      </defs>
    </svg>
    <div
      className="star-container"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      style={
        Object {
          "cursor": undefined,
          "display": "inline-block",
          "paddingLeft": undefined,
          "paddingRight": "5px",
          "position": "relative",
          "verticalAlign": "middle",
        }
      }
    >
      <svg
        className="widget-svg"
        style={
          Object {
            "height": "20px",
            "transform": undefined,
            "transition": "transform .2s ease-in-out",
            "width": "20px",
          }
        }
        viewBox="0 0 51 48"
      >
        <path
          className="star"
          d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          style={
            Object {
              "fill": "gold",
              "transition": "fill .2s ease-in-out",
            }
          }
        />
      </svg>
    </div>
    <div
      className="star-container"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      style={
        Object {
          "cursor": undefined,
          "display": "inline-block",
          "paddingLeft": "5px",
          "paddingRight": "5px",
          "position": "relative",
          "verticalAlign": "middle",
        }
      }
    >
      <svg
        className="widget-svg"
        style={
          Object {
            "height": "20px",
            "transform": undefined,
            "transition": "transform .2s ease-in-out",
            "width": "20px",
          }
        }
        viewBox="0 0 51 48"
      >
        <path
          className="star"
          d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          style={
            Object {
              "fill": "gold",
              "transition": "fill .2s ease-in-out",
            }
          }
        />
      </svg>
    </div>
    <div
      className="star-container"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      style={
        Object {
          "cursor": undefined,
          "display": "inline-block",
          "paddingLeft": "5px",
          "paddingRight": "5px",
          "position": "relative",
          "verticalAlign": "middle",
        }
      }
    >
      <svg
        className="widget-svg"
        style={
          Object {
            "height": "20px",
            "transform": undefined,
            "transition": "transform .2s ease-in-out",
            "width": "20px",
          }
        }
        viewBox="0 0 51 48"
      >
        <path
          className="star"
          d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          style={
            Object {
              "fill": "gold",
              "transition": "fill .2s ease-in-out",
            }
          }
        />
      </svg>
    </div>
    <div
      className="star-container"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      style={
        Object {
          "cursor": undefined,
          "display": "inline-block",
          "paddingLeft": "5px",
          "paddingRight": "5px",
          "position": "relative",
          "verticalAlign": "middle",
        }
      }
    >
      <svg
        className="widget-svg multi-widget-selected"
        style={
          Object {
            "height": "20px",
            "transform": undefined,
            "transition": "transform .2s ease-in-out",
            "width": "20px",
          }
        }
        viewBox="0 0 51 48"
      >
        <path
          className="star"
          d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          style={
            Object {
              "fill": "url('#starGrad691250465249835')",
              "transition": "fill .2s ease-in-out",
            }
          }
        />
      </svg>
    </div>
    <div
      className="star-container"
      onClick={null}
      onMouseEnter={null}
      onMouseLeave={null}
      style={
        Object {
          "cursor": undefined,
          "display": "inline-block",
          "paddingLeft": "5px",
          "paddingRight": undefined,
          "position": "relative",
          "verticalAlign": "middle",
        }
      }
    >
      <svg
        className="widget-svg"
        style={
          Object {
            "height": "20px",
            "transform": undefined,
            "transition": "transform .2s ease-in-out",
            "width": "20px",
          }
        }
        viewBox="0 0 51 48"
      >
        <path
          className="star"
          d="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
          style={
            Object {
              "fill": "rgb(203, 211, 227)",
              "transition": "fill .2s ease-in-out",
            }
          }
        />
      </svg>
    </div>
  </div>,
]
`;
